cleanup(angular): consolidate and improve e2e-angular-core tests

leosvelperez opened the pull request:
## What it does

- Remove some duplicated work:
	- Multiple tests covering the same task runs with no added value
	- Generation of new projects when the existing ones could be reused
- Remove an invalid test suite that's testing something we don't target to support `ng-cli.test.ts`
- Improve ports management/cleanup
- Add caching to creating Angular CLI workspaces

~With the changes in the PR, the time is reduced, but it's still insufficient to meet the target. The last run did end up below the 15-minute mark (14m 27s), but there's currently a test case that's temporarily disabled which will bring the time up once enabled. In a separate branch/PR I'll explore splitting the test project.~

The latest run took 12m 56s, so we seem to meet the target now. We'll keep an eye on it.

meeroslav commented on e2e/utils/process-utils.ts:
```diff
@@ -39,3 +39,19 @@ export async function killPorts(port?: number): Promise<boolean> {
+      await killPort(port);
+    }
+  } catch (err) {
+    expect(err).toBeFalsy();
```
We want to catch an error, print it out and fail test rather than have exception blow up, since it's better for the readability and subsequent debugging.

This way failed tests have the same output format as those that had an exception during the run.

Apart from that, there is no other reason. That assertion could be anything as long as it reports a mismatch and shows an entire error with the stack trace.

meeroslav replied:
If you find a better way to do this, feel free to improve it everywhere

leosvelperez replied:
Sounds good, I'll check whether it makes sense to do something else, but it's not a priority anyways. This is just a cleanup, not an actual assertion or the focus of the tests.

barbados-clemens replied:
cool, makes sense just wanted to make sure it wasn't supposed to be changed/updated for any reason.

meeroslav commented on e2e/angular-core/src/angular-linting.test.ts:
```diff
@@ -7,73 +7,63 @@ import {
-    beforeAll(() => newProject());
-    afterAll(() => cleanupProject());
+describe('linting', () => {
+  beforeAll(() => newProject());
```
Can we combine linting with some other suite so that we can reuse the same `newProject` and potentially also the same `app` and `lib`?

leosvelperez commented:
@meeroslav thanks for the tips! Those are great and I will work on applying them.

As mentioned in the PR description, I'll split the tests on a separate PR. I also want to check the `angular-extensions` project and split them more evenly, so I'll keep that separate.

meeroslav commented:
Please note that each test project splitting leads to a multiplication of git checkout, yarn install, verdaccio publish... 

This should truly be only the last resort and only makes sense for test projects that take longer than 10 minutes since the preparation part often takes up to 3-4 minutes.

leosvelperez commented:
> Please note that each test project splitting leads to a multiplication of git checkout, yarn install, verdaccio publish...
> 
> This should truly be only the last resort and only makes sense for test projects that take longer than 10 minutes since the preparation part often takes up to 3-4 minutes.

That is why I'm keeping it as a separate change and don't want to just extract the MF tests into a project. If it comes to splitting the projects, it would be better to split the current two projects into three to have a more even split.

leosvelperez commented:
@meeroslav I addressed the changes you requested except:

- Splitting Module Federation tests into a separate test project, as mentioned previously, splitting the current Angular e2e tests is something I'll explore in a different PR
- Merging the Angular Config test suite with another. We're keeping this one separate. Its setup is not that of a regular workspace because we use `angular.json` with v1 format. It's not impossible to merge it with another suite, but I rather not do it. It'd be messy with the setup and cleanup between tests and affect the readability of the test suite.

Additionally, I removed an additional test suite I realized was testing something we don't really target to support, so we shouldn't be testing it.

meeroslav reviewed (approved):
Great work! Thank you!
